packages/components/table/src/use-table.ts

```diff
@@ -188,7 +187,10 @@ export function useTable<T extends object>(originalProps: UseTableProps<T>) {
 
   const {collection} = state;
 
-  const {gridProps} = useReactAriaTable<T>({...originalProps, layout: layoutNode}, state, domRef);
+  // Exclude the layout prop because it has a name conflict and is deprecated in useTable.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const {layout, ...otherOriginalProps} = originalProps;
+  const {gridProps} = useReactAriaTable<T>({...otherOriginalProps}, state, domRef);
```

Comment:
`layout` (`layoutNode`) has been deprecated, so I'm not passing it. If needed, you can use `layoutDelegate` instead. This is a breaking change, but since the `layoutNode` prop is quite a specific use case, I don't think it will be a major issue.
https://github.com/adobe/react-spectrum/pull/6518/files